portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java

```diff
@@ -716,6 +716,36 @@ public FileEntry checkOutFileEntry(
 		return fileEntry;
 	}
 
+	@Override
+	public FileEntry copyFileEntry(
+			long fileEntryId, long destinationFolderId,
+			long destinationRepositoryId, ServiceContext serviceContext)
+		throws PortalException {
+
+		Repository sourceRepository = repositoryProvider.getFileEntryRepository(
+			fileEntryId);
+
+		return copyFileEntry(
+			getRepository(destinationRepositoryId),
+			sourceRepository.getFileEntry(fileEntryId), destinationFolderId,
+			serviceContext);
+	}
+
+	public FileShortcut copyFileShortcut(
```

Comment:
Shouldn't there be an `@Override` here?

Reply:
Yes! Strange the source formatter doesn't complaint.

Reply:
done